### PR TITLE
Store the original test, check and subresult outcome in results

### PR DESCRIPTION
**Pull request by happz**
Pull Request Checklist

* [x] implement the feature
* [x] write the documentation
* [x] update the specification
* [x] mention the version
* [x] include a release note

### CODE REVIEW

**Review by KwisatzHaderach — APPROVED** [review]

**Review by therazix — APPROVED** [review]

**Review by seberm — APPROVED** [review]

**Review by martinhoyer — APPROVED** [review]
I wish I saw this before working on it in https://github.com/teemtee/tmt/pull/3239 lol

**Comment by martinhoyer** [comment]
btw, in results.py 
```
        # Extend existing note or set a new one
        if self.note:
            self.note += f', original result: {self.result.value}'
        elif self.note is None:
            self.note = f'original result: {self.result.value}'
```
shouldn't it be `self.original_result.value`?

**Comment by happz** [comment]
> btw, in results.py
> 
> ```
>         # Extend existing note or set a new one
>         if self.note:
>             self.note += f', original result: {self.result.value}'
>         elif self.note is None:
>             self.note = f'original result: {self.result.value}'
> ```
> 
> shouldn't it be `self.original_result.value`?

At this moment, `self.result` and `self.original_result` should be the same value, I'm looking at code and `self.result` is modified a few lines later.

**Review by psss — CHANGES_REQUESTED** [review]

File: tmt/schemas/results.yaml [hunk 1]
```diff
@@ -29,6 +29,9 @@ definitions:
       result:
         $ref: "/schemas/common#/definitions/result_outcome"
 
+      original_result:
```
[the remark this answers is not in the document]
Comment (happz) [reply tmt/schemas/results.yaml]: Correct, fixed in 2ca36d83

**Review by psss — APPROVED** [review]
Looks good, thanks!

**Comment by psss** [comment]
Red tests are irrelevant failures, merging.

**Pull request merged**